dhall/src/Dhall/Freeze.hs

```diff
@@ -158,15 +158,36 @@ freeze outputMode input scope intent characterSet censor = do
     let freezeFunction = freezeScope directory
 
     let cache
+            -- This case is necessary because `transformOf` is a bottom-up
+            -- rewrite rule.   Without this rule, if you were to transform a
```

Comment:
Tangential: Are there any top-down rewrite rules that we could use instead? I believe we have to apply similar workarounds in other places.

Reply:
> Are there any top-down rewrite rules that we could use instead?

I assume the types probably wouldn't work out…

Reply:
There aren't any that I know of.